### ISSUE TITLE
Setting BodyType::STATIC before calling updateMassPropertiesFromColliders() disables collision detection

### ISSUE THREAD

**Issue by acr** [DanielChappuis/reactphysics3d#225]
How to reproduce:

```
reactphysics3d::RigidBody* rigid_body = world->createRigidBody(transform);

rigid_body->addCollider(shape, reactphysics3d::Transform::identity());
rigid_body->setType(reactphysics3d::BodyType::STATIC); // In this order, CD does not work
rigid_body->updateMassPropertiesFromColliders();       // against this static object.
```
^^ `BodyType::DYNAMIC` objects will not collide with this object. But if the following order of operations is done instead they will:

```
reactphysics3d::RigidBody* rigid_body = world->createRigidBody(transform);

rigid_body->addCollider(shape, reactphysics3d::Transform::identity());
rigid_body->updateMassPropertiesFromColliders();       // Changing the order of these
rigid_body->setType(reactphysics3d::BodyType::STATIC); // operations makes CD work.
```

**Comment by BabyBobbyBoy** [comment]
I also experienced this bug.  Thank you ACR for giving a temporary workaround.

**Comment by DanielChappuis** [comment]
Thanks a lot for taking the time to report this issue. I think it's already fixed in the 'develop' branch because I cannot reproduce it on this branch.

This fix will be part of the next version of the library that will be released soon.

**Comment by acr** [comment]
Thank you!

**Comment by BabyBobbyBoy** [comment]
Thank you Daniel for looking into this.  Also thank you for writing the ReactPhysics3D engine.  I have been impressed by your implementation.

**Comment by DanielChappuis** [comment]
This issue is now fixed in version [v0.9.0](https://github.com/DanielChappuis/reactphysics3d/releases/tag/v0.9.0) that has just been released.
